mithril.js

```diff
@@ -1488,18 +1488,26 @@
 		// m.route()
 		if (arguments.length === 0) return currentRoute
 		// m.route(el, defaultRoute, routes)
-		if (arguments.length === 3 && isString(arg1)) {
+		// m.route(el, defaultComponent, routes)
+		if (arguments.length === 3 && (isString(arg1) || isObject(arg1))) {
 			redirect = function (source) {
 				var path = currentRoute = normalizeRoute(source)
 				if (!routeByValue(root, arg2, path)) {
-					if (isDefaultRoute) {
-						throw new Error("Ensure the default route matches " +
-							"one of the routes defined in m.route")
+					// defaultRoute
+					if(isString(arg1)) {
```

Comment:
@ludbek 

Not saying there's anything fundamentally wrong with the way you are doing it, but I'm just curious why `isDefaultRoute` no longer works here.